source/grid/tria_info_cache.cc

```diff
@@ -0,0 +1,56 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/grid/tria_info_cache.h>
+#include <deal.II/grid/grid_tools.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+template<int dim, int spacedim>
+TriangulationInfoCache<dim,spacedim>::TriangulationInfoCache(const Triangulation<dim, spacedim> &tria,
+    const TriangulationInfoCacheFlags &flags,
+    const Mapping<dim, spacedim> &mapping) :
+  tria(&tria),
+  flags(flags),
+  mapping(&mapping)
+{
+  tria->signals.any_change.connect([&]()
+  {
+    update();
+  });
```

Comment:
Should we also call `update` here as soon as the object is constructed? This way it is immediately useful.

Reply:
Yes, done below this comment.